### PR TITLE
🚀 improve file-box module

### PR DESCRIPTION
**Pull request by wj-Mcat**
There are some functions that should be improve to support more features.  

- [x] `to_file`
- [x] `FileBoxType`  enum definition problems
- [x] `ready` the name&mimetype from remote type
- [x] `to_file` save the `FileBox` to the local position.
- [x] `from_base64` name params is required
- [x] `from_data_url` can transfer `data_url` data to the FileBox type
- [x] add unit test 
- [ ] `from_buffer` and `from_stream` have not encountered in our test.
- [ ] `to_buffer` and `to_stream` have not encountered in our test.

### CODE REVIEW

File: src/wechaty_puppet/file_box/file_box.py [hunk 1]
```diff
@@ -92,9 +93,40 @@ def type(self) -> FileBoxType:
...
+
+            headers in requests package doesn't contains attribute:
+                content-disposition, so we need to change the way to sync
+                file-name from url type
+
+            I find the better way to extract file name from url:
+                https://stackoverflow.com/questions/18727347/how-to-extract-a-
+                filename-from-a-url-append-a-word-to-it/18727481#18727481
+        """
+        if self.boxType is not FileBoxType.Url:
```
Comment (univerone) [inline src/wechaty_puppet/file_box/file_box.py:122]: `self.boxTyle` is a number, which can not be compared to FileBoxType.Url

Comment (wj-Mcat) [reply src/wechaty_puppet/file_box/file_box.py:122]: Yes, it's my fault to write this type checking code.  I think the correct type checking is :
```python
file_box_type = self.type()
if file_box_type != FileBoxType.Url:
    raise TypeError('type <{0}> is not remote'.format(
        file_box_type.name))
```
Glad to see that you can review  so detailed checking, please go ahead to all pr.

File: src/wechaty_puppet/file_box/file_box.py [hunk 2]
```diff
@@ -109,12 +143,39 @@ def to_json_str(self) -> str:
...
+        file_path = self.name if file_path is None else file_path
+
+        if os.path.exists(file_path) and not overwrite:
+            raise FileExistsError(f'FileBox.toFile(${file_path}): file exist. '
+                                  f'use FileBox.toFile(${file_path}, true) to '
+                                  f'force overwrite.')
+        file_box_type = self.type()
+
+        if file_box_type == FileBoxType.Buffer:
+            with open(file_path, 'wb+', encoding='utf-8') as f:
```
Comment (univerone) [inline src/wechaty_puppet/file_box/file_box.py:165]: this line will raise` internal error <binary mode doesn't take an encoding argument>`

Comment (wj-Mcat) [reply src/wechaty_puppet/file_box/file_box.py:165]: Oh yes, if the mode of IoStream is 'wb', we should not set encoding argument.

**Comment by wj-Mcat** [comment]
Thanks for so carefully test.  Please go ahead to push the process of code review. @univerone

**Pull request merged**